contracts/Consensus.sol

```diff
@@ -123,4 +125,21 @@ contract Consensus is ConsensusUtils {
     require(_amount >= getMinValidatorFee());
     _setValidatorFee(msg.sender, _amount);
   }
+
+  /**
+  * @dev Function to be called by jailed validator, in order to be released from jail
+  */
+  function unJail() external onlyJailedValidator {
+    require(getReleaseBlock(msg.sender) <= getCurrentCycleEndBlock());
+
+    _removeFromJail(msg.sender);
+  }
+
+  /**
+  * @dev Function to be called by current validators to be dropped from the next cycle in order to perform maintenance 
+  */
+  function maintenance() external onlyValidator {
```

Comment:
You add the validator to jail here, right?
Can you explain how it's gonna used by the validators

Reply:
Yes so flagging for maintenance moves them to the jailed list but doesn't impose a strike and they can signal to be released at the start of the next cycle (no min jail time)